mitiq/shadows/quantum_processing.py

```diff
@@ -0,0 +1,199 @@
+from mitiq.interface.mitiq_qiskit.conversions import to_qiskit
+from mitiq.interface.mitiq_qiskit.qiskit_utils import (
+    sample_bitstrings as qiskit_sample_bitstrings,
+)
+
+
+def generate_random_pauli_strings(
+    num_qubits: int, num_strings: int
+) -> List[str]:
+    """Generate a list of random Pauli strings.
+
+    Args:
+        num_qubits: The number of qubits in the Pauli strings.
+        num_strings: The number of Pauli strings to generate.
+
+    Returns:
+        A list of random Pauli strings.
+    """
+
+    # Sample random Pauli operators uniformly from X, Y, Z
+    unitary_ensemble = ["X", "Y", "Z"]
+    paulis = np.random.choice(unitary_ensemble, (num_strings, num_qubits))
+    return ["".join(pauli) for pauli in paulis]
+
+
+# attach measurement gates to list of circuits
+def get_rotated_circuits(
+    circuit: cirq.Circuit,
+    pauli_strings: List[str],
+    add_measurements: bool = True,
+) -> List[cirq.Circuit]:
+    """Returns a list of circuits that are identical to the input circuit,
+    except that each one has single-qubit Clifford gates followed by
+    measurement gates that are designed to measure the input
+    Pauli strings in the Z basis.
+
+    Args:
+        circuit: The circuit to measure.
+        pauli_strings: The Pauli strings to measure in each output circuit.
+        add_measurements: Whether to add measurement gates to the circuit.
+
+    Returns:
+         The list of circuits with rotation and measurement gates appended.
+    """
+    qubits = sorted(list(circuit.all_qubits()))
+    num_qubits = len(qubits)
+    rotated_circuits = []
+    for pauli_string in pauli_strings:
+        assert len(pauli_string) == num_qubits, (
+            f"Pauli string must be same length as number of qubits, "
+            f"got {len(pauli_string)} and {num_qubits}"
+        )
+        rotated_circuit = circuit.copy()
+        for i, pauli in enumerate(pauli_string):
+            qubit = qubits[i]
+            # Pauli X measurement is equivalent to H plus a Z measurement
+            if pauli == "X":
+                rotated_circuit.append(cirq.H(qubit))
+            # Pauli X measurement is equivalent to S^-1*H plus a Z measurement
+            elif pauli == "Y":
+                rotated_circuit.append(cirq.S(qubit) ** -1)
+                rotated_circuit.append(cirq.H(qubit))
+            # Pauli Z measurement
+            else:
+                assert (
+                    pauli == "Z"
+                ), f"Pauli must be X, Y, Z. Got {pauli} instead."
+        if add_measurements:
+            # append measurement gates
+            rotated_circuit.append(cirq.measure(*qubits))
+        # append to list of circuits
+        rotated_circuits.append(rotated_circuit)
+    return rotated_circuits
+
+
+# Stage 1 of Classical Shadows: Measurements
+def get_z_basis_measurement(
+    circuit: cirq.Circuit,
+    n_total_measurements: int,
+    sampling_function: Union[str, Callable[..., MeasurementResult]] = "cirq",
```

Comment:
For consistency with all Mitiq techniques, I suggest to call this function `executor` and to avoid having a default noiseless executor.   A design principle of Mitiq is that users are supposed to write down their own `Executor` objects or `execute` functions before using Mitiq. In all Mitiq techniques, the definition of the `executor` argument should not be internally given by Mitiq, but externally given by users.

```suggestion
    executor: Union[Executor, Callable[[QPROGRAM], MeasurementResult]]
```
or `executor: Union[Executor, Callable[[cirq.Circuit], MeasurementResult]]` if we only consider Cirq circuits.

Reply:
Thanks! I'll call it random_pauli_measurement, how does this sound?

Reply:
have been updated in the new version.